src/views/Rules/ListRuleRevisions/index.jsx

```diff
@@ -99,8 +101,27 @@ function ListRuleRevisions(props) {
     });
   };
 
-  // TODO: Add logic to restore a revision
-  const handleRestoreClick = () => {};
+  const handleRestoreClick = row => async () => {
+    const rowData = clone(row);
+
+    // We only want the actual rule data from the old revision,
+    // not the metadata.
+    delete rowData.change_id;
+    delete rowData.changed_by;
+    delete rowData.timestamp;
+    delete rowData.data_version;
+    const { error, data } = await addSC({
+      change_type: 'update',
+      when: new Date().getTime() + 5000,
+      data_version: revisions[0].data_version,
+      ...rowData,
+    });
+
+    if (!error) {
+      props.history.push(`/rules#scId=${data.data.sc_id}`);
+    }
+  };
```

Comment:
Should we show a confirmation dialog when restoring? Are there any bad consequences if a user clicks on restore by mistake?

Reply:
All of the important channels are protected by signoffs, so nothing bad can happen directly. But nonetheless, I think this is a good idea. I can see this button getting clicked by accident when trying to click `View`.